pycbc/events/coinc.py

```diff
@@ -240,6 +240,90 @@ def time_coincidence(t1, t2, window, slide_step=0):
     return idx1.astype(numpy.uint32), idx2.astype(numpy.uint32), slide.astype(numpy.int32)
 
 
+def time_multi_coincidence(times, slide_step=0, slop=.003,
+                           pivot='H1', fixed='L1'):
+    """ Find multi detector concidences.
+
+    Parameters
+    ----------
+    times: dict of numpy.ndarrays
+        Dictionary keyed by ifo of the times of each single detector trigger.
+    slide_steop: float
+        The interval between time slides
+    slop: float
+        The amount of time to add to the TOF between detectors for coincidence
+    pivot: str
+        ifo used to test coincidence against in first stage
+    fixed: str
+        the other ifo used in the first stage coindicence which we'll use
+        as a fixed time reference for coincident triggers. All other detectors
+        are time slid by being fixed to this detector.
+    """
+    from pycbc.detector import Detector
+
+    # pivots are used to determine standard coincidence triggers, we then
+    # pair off additional detectors to those.
+    def win(ifo1, ifo2):
+        d1 = Detector(ifo1)
+        d2 = Detector(ifo2)
+        return d1.light_travel_time_to_detector(d2) + slop
+
+    # Find coincs first between the two fully timselide detectors
+    pivot_id, fix_id, slide = time_coincidence(times[pivot], times[fixed],
+                                               win(pivot, fixed),
+                                               slide_step=slide_step)
+
+    # additional detectors are conceptually stitched to the 'fixed' one.
+    # Each trigger in an additional detector must be concidence with an
+    # existing coincident one. All times moved to 'fixed' relative time
+    fixed_time = times[fixed][fix_id]
+    pivot_time = times[pivot][pivot_id] - slide_step * slide
+
+    ctimes = {fixed: fixed_time, pivot:pivot_time}
+    ids = {fixed:fix_id, pivot:pivot_id}
+
+    dep_ifos = [ifo for ifo in times.keys() if ifo != fixed and ifo != pivot]
+    for ifo1 in dep_ifos:
```

Comment:
Could make an argument for the code starting here to go in a separate function??

Reply:
I think if we ever decide to do something more elaborate that involves scrambling ime with respect to the 'fixed' ifo in some way, then sure.